-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33803/22827

ERROR: Build errors found during clang-tidy run.
```
DQMServices/Components/plugins/DQMGenericClient.cc:176:12: error: use of undeclared identifier 'fitFcn' [clang-diagnostic-error]
    delete fitFcn;
           ^
Suppressed 1852 warnings (1851 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/DQMServices/Components/plugins/DQMGenericClient.cc
+++ b/DQMServices/Components/plugins/DQMGenericClient.cc
@@ -157,7 +157,7 @@ public:
     TH1::AddDirectory(true);
     // ... create your hists
     TH2F* h = me->getTH2F();
-    TF1* fitFcn = new TF1("gaus", "gaus",h->GetYaxis().GetXmin(),h->GetYaxis().GetXmax());
+    TF1* fitFcn = new TF1("gaus", "gaus", h->GetYaxis()->GetXmin(), h->GetYaxis()->GetXmax());
     h->FitSlicesY(fitFcn, 0, -1, 0, "QNR SERIAL");
     string name(h->GetName());
     h0 = (TH1*)gDirectory->Get((name + "_0").c_str());

--- a/DQMServices/Components/plugins/DQMGenericClient.cc
+++ b/DQMServices/Components/plugins/DQMGenericClient.cc
@@ -165,6 +165,7 @@ public:
     h2 = (TH1*)gDirectory->Get((name + "_2").c_str());
     h3 = (TH1*)gDirectory->Get((name + "_chi2").c_str());
     TH1::AddDirectory(oldAddDir);
+    delete fitFcn;
   }
 
   /// Destructor
@@ -173,7 +174,6 @@ public:
     delete h1;
     delete h2;
     delete h3;
-    delete fitFcn;
   }
   /// Fill the ME with the mean value of the gaussian fit in each slice
   void getFittedMean(MonitorElement* me) {

--- a/DQMServices/Components/plugins/DQMGenericClient.cc
+++ b/DQMServices/Components/plugins/DQMGenericClient.cc
@@ -157,7 +157,7 @@ public:
     TH1::AddDirectory(true);
     // ... create your hists
     TH2F* h = me->getTH2F();
-    TF1* fitFcn = new TF1("gaus", "gaus");
+    TF1* fitFcn = new TF1("gaus", "gaus",h->GetYaxis().GetXmin(),h->GetYaxis().GetXmax());
     h->FitSlicesY(fitFcn, 0, -1, 0, "QNR SERIAL");
     string name(h->GetName());
     h0 = (TH1*)gDirectory->Get((name + "_0").c_str());

